.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,59 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+    # only run if files in src/ have been changed
+    paths:
+      - 'src/**'
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+    paths:
+      - 'src/**'
```

Comment:
If we build it always and not just on `src` changes, we can require this check to be clean for all PRs before merging